src/reformat/rgb_impl.rs

```diff
@@ -210,31 +210,39 @@ fn compute_rgb(y: f32, cb: f32, cr: f32, has_color: bool, mode: Mode) -> (f32, f
     )
 }
 
-fn clamped_pixel(
-    depth: u8,
-    // Technically, these two are options since one of them will always be Ok.
-    row: &AvifResult<&[u8]>,
-    row16: &AvifResult<&[u16]>,
-    index: usize,
-    max_channel: u16,
-) -> u16 {
-    if depth == 8 {
-        row.unwrap()[index] as u16
-    } else {
-        min(max_channel, row16.unwrap()[index])
+enum Row<'a> {
+    Absent,
```

Comment:
this seems like it should be an `Optional` instead of being an enum value. Since it basically represents `None`.